Licensing docs: Add docs about usage billing option (#31267)
grafana/grafana: docs/sources/enterprise/license-restrictions.md

When the number of maximum active viewers or editors is reached, Grafana displays a warning banner.

### Usage billing

> It is possible to turn on usage billing in enterprise licensing to have unlimited number of user limits. When usage billing is on, Grafana will not enforce active user limits and instead will charge for active users above the limit, according to the customer's contract. 

Reviewer: ```suggestion
It is possible to turn on usage billing in Enterprise licensing in order to have an unlimited number of user limits. When usage billing is on, Grafana does not enforce active user limits. Instead, Grafana charges for active users above the limit, according to your customer contract.
```

> It is possible to turn on usage billing in Enterprise licensing in order to have an unlimited number of user limits. When usage billing is on, Grafana does not enforce active user limits. Instead, Grafana charges for active users above the limit, according to your customer contract.

Reviewer 2: ```suggestion
It is possible to turn on usage billing in Enterprise licensing in order to have an unlimited number of user limits. When usage billing is on, Grafana does not enforce active user limits. Instead, Grafana Labs charges for active users above the limit, according to your customer contract.
```

@osg-grafana I'm not sure about this, but I think we might want to refer to the billing entity as "Grafana Labs" (or something like "your Grafana vendor", but I don't think that's common enough that we need to care about that at the moment)?

> The usage billing must be agreed with Grafana Labs and is only available to customers who have automated license token renewal enabled.

Reviewer: ```suggestion
The usage billing must be agreed upon with Grafana Labs and is only available to customers who have automated license-token renewal enabled.
```

Reviewer: Good enough atm, but sounds like pricing speak. Can we simplify this? I need to eat lunch, so brain on empty atm.

Author: Curious why the dash for `license-token`?

Reviewer: Looks like a compound adjective.

Reviewer 3: My take:

"Usage billing must be agreed upon with Grafana Labs, and it is only available if you have turned on automatic license token renewal."

I say "automatic license token renewal" because it's the name of a specific feature. 

"automatic license token renewal" should link to docs for that feature, but it looks like those docs do not yet exist. Not to pull the string too hard here, but has anyone looked at docs for automatic license token renewal? I don't even see the config option documented. @sakjur this might have been part of #835

Reviewer 2: #30217

Reviewer 3: Thanks, I didn't see that there is an enterprise-specific configuration page. That's a little confusing; perhaps something to consider as we structure docs pages @osg-grafana

Reviewer 2: 👆 I agree and I have brought this up with @osg-grafana before as something we should consider. It's hard to say what's the right option here (how do we differentiate two products which are _almost but critically not quite_ the same), but I think we'd need some clearer and standardized way to flag enterprise-only configuration options intermixed with the regular configuration page.

Reviewer 3: Yes agreed. There is currently debate raging on this in a series of workshops. My opinion is that we should have a filter on the page for Grafana "edition" (OSS, Pro, Advanced, Enterprise) and use tags to highlight premium features. Gitlab does this: https://docs.gitlab.com/ee/user/group/saml_sso/

Reviewer: I am going to leave this discussion for a planning session outside a PR.

Reviewer: fwiw, looks like GitLab uses Nanoc, and Denis (founder of Nanoc) is a friend of mine.

Sometimes it is useful to log in to an account from multiple locations concurrently. With Grafana Enterprise, accounts are limited to three concurrent sessions.

### Usage billing

It is possible to turn on usage billing in Enterprise licensing in order to have an unlimited number of user limits. When usage billing is on, Grafana does not enforce active user limits. Instead, Grafana Labs charges for active users above the limit, according to your customer contract. 

> Usage billing must be agreed upon with Grafana Labs, and it is only available if Grafana Enterprise is configured to be [automatically refreshed](https://grafana.com/docs/grafana/latest/enterprise/enterprise-configuration/#auto_refresh_license).

Reviewer 2: Trying to squeeze some ambiguitiy.

```suggestion
Usage billing must be agreed upon with Grafana Labs, and it is only available if Grafana Enterprise is configured to [automatically refresh the license](https://grafana.com/docs/grafana/latest/enterprise/enterprise-configuration/#auto_refresh_license).
```

Given a limit of three sessions, the longest inactive session is signed out of when a fourth person signs in to the same account.

### Usage billing

> You can turn on usage billing in Enterprise licensing in order to have an unlimited number of active users. When usage billing is enabled, Grafana does not enforce active user limits. Instead, we charge for active users above the limit, according to your customer contract. 

Reviewer 2: I'm not sure the latest "You can" change is semantically correct, as that is not an action taken by the customer. Wdyt? 👇

```suggestion
You can request usage billing in Enterprise licensing in order to have an unlimited number of active users. When usage billing is enabled, Grafana does not enforce active user limits. Instead, we charge for active users above the limit, according to your customer contract. 
```

Author: Yeah good point, this though sounds a bit odd
> You can request usage billing in order to have an unlimited number of active users

What about
>You can request usage billing to be turned on in order to have an unlimited number of active users

and add a link to wherever they can request it.

Author: cc @oddlittlebird what do you think about the suggestion?

Reviewer 4: Works for me. I just try to replace the "it is possible" construction wherever I see it.

Reviewer 5: Another suggestion:
"You can request Grafana Labs to enable billing to allow an unlimited number of active users ..."

Author: There have been multiple discussions around the wording "turn on" vs "enable". I honestly lost at one point what was the consensus, but I know that in the UI we shod the flag as "On" or "Off".  So perhaps mentioning "to be turned on" would give better context to customers about what we are talking about.

Author: @achatterjee-grafana I've incorporated your feedback, hope this looks good now. If there are no objections I'd like to get this merged today.

Thanks everyone for contributions! 💯

> It is possible to turn on usage billing in Enterprise licensing in order to have an unlimited number of active users. When usage billing is enabled, Grafana does not enforce active user limits. Instead, we charge for active users above the limit, according to your customer contract. 

Reviewer 4: ```suggestion
You can turn on usage billing in Enterprise licensing in order to have an unlimited number of active users. When usage billing is enabled, Grafana does not enforce active user limits. Instead, we charge for active users above the limit, according to your customer contract. 
```
